switch_model/generators/core/dispatch.py

```diff
@@ -273,6 +274,35 @@ def init_gen_availability(m, g):
     mod.have_minimal_gen_max_capacity_factors = BuildCheck(
         mod.VARIABLE_GEN_TPS,
         rule=lambda m, g, t: (g,t) in m.VARIABLE_GEN_TPS_RAW)
+    
+    # Generate a warning if the input files specify timeseries for renewable
+    # plant capacity factors that extend beyond the expected lifetime of the
+    # plant. This could be caused by simple logic to build input files, or
+    # could indicate that the user expects those plants to operate longer
+    # than indicated.
+    def _warn_on_extra_VARIABLE_GEN_TPS(m):
+        extra_indexes = set(m.VARIABLE_GEN_TPS_RAW) - set(m.VARIABLE_GEN_TPS)
+        num_impacted_generators = len(set([g for g,t in extra_indexes]))
+        extraneous = {g: [] for (g,t) in extra_indexes}
+        for (g,t) in extra_indexes:
+            extraneous[g].append(t)
+        pprint = "\n".join(
+            "* {}: {}".format(g, tps) for g, tps in extraneous.items())
+        warning_msg = (
+            "{} renewable generators have capacity factors specified "
+            "in timepoints after they are slated for retirement. This "
+            "could indicate a benign issue where the process that built "
+            "the dataset used simplified logic and/or didn't know the "
+            "scheduled retirement date. If you expect those datapoints to "
+            "be useful, then those plants need longer lifetimes (or "
+            "options to build new capacity when the old capacity reaches "
+            "the provided end-of-life date).\n".format(num_impacted_generators))
+        if extra_indexes:
+            logging.warning(warning_msg)
+            logging.info("Plants with extra timepoints:\n{}".format(pprint))
+        return(True)
+    mod.warn_on_extra_VARIABLE_GEN_TPS = BuildCheck(
+        rule=_warn_on_extra_VARIABLE_GEN_TPS)
```

Comment:
In https://github.com/switch-model/switch/pull/115#discussion_r341685397 I recommended using `warning` as our default logging level, and defining the `info` level as being equivalent to our current `--verbose` flag. For the cases shown here--where we allow data for extra timepoints but it's not considered good practice for purists--I would show the messages at the `debug` level, not the `info` or `warning` level. If we've decided that this behavior is OK, then we shouldn't hassle the user with warning messages every time they run the model.

Reply:
This is now addressed in https://github.com/switch-model/switch/pull/115/commits/4642de038126b9747cd1ad4f9b6638338b82a7e0. I relented a little and (at the cost of some extra fanciness) give a general message at the `info` level that there are some unused entries in `variable_capacity_factors.csv` (not a warning, just a friendly note), and give more details at the `debug` level. This may be a good template for the future, or may be unnecessarily complex with an unclear line between warning, info and debug.